### PR TITLE
Added documentation for RegexMatch

### PR DESCRIPTION
**Pull request by CalLavicka**

### CODE REVIEW

**Review by melodeeli98 — APPROVED** [review]

**Review by ermay12 — CHANGES_REQUESTED** [review]
wait until i merge my pr first.

**Review by alex-stanescu — APPROVED** [review]

**Review by ermay12 — APPROVED** [review]

**Pull request closed without merging**
